### DISCUSSION TITLE
Re-write model with current YOLOv5/PyTorch versions?

### DISCUSSION THREAD

**Discussion by agentmorris** [ultralytics/yolov5#9138] [General]
I have a YOLOv5 model that was trained a few months ago, and due to [issue 6948](https://github.com/ultralytics/yolov5/issues/6948), I cannot run inference with current PyTorch builds.  I would like to allow inference with current PyTorch and YOLOv5 builds, in particular to take advantage of recently-added M1 support.

My understanding, based the discussion of that issue (in particular @glenn-jocher's [reply](https://github.com/ultralytics/yolov5/issues/6948#issuecomment-1072291513), is that updating the inference environment is not sufficient; the model needs to be re-written.  Experiments suggest that fine-tuning on new data achieves the desired result, but may modify the weights, which we don't want in this case; it's important that the weights not change at all.

Is there a recommended way of "fine-tuning" without adding any new data, i.e. just re-writing the existing weights in a new YOLOv5/PyTorch environment?

Thanks!

-Dan

**Comment by glenn-jocher** [comment]
@agentmorris I'm unable to produce any backwards compatibility issues with older models running inference in master. Here is a v6.0 model trained in October 2021 running inference with today's master:

```bash
!python detect.py --weights https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5s.pt
```

<img width="1259" alt="Screenshot 2022-08-29 at 19 45 15" src="https://user-images.githubusercontent.com/26833433/187265201-b8d6f159-4526-4ec0-ae46-430eaf93125a.png">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by agentmorris** [comment]
Thanks!  You're right, it does seem to be working with the latest YOLOv5 and the latest PyTorch.  Unless I'm losing my mind, based on my own experience and @glenn-jocher's reply to issue 6498, it seems that there was a period where the "upsample" incompatibility had been fixed, but running old models (from before the incompatibility existed) still required re-training.  But now the incompatibility appears to be... even more fixed, such that even old models work.

So, sorry for the false alarm, and thanks for the quick reply!

**Comment by agentmorris** [comment]
Interestingly, although as per above I confirmed that this issue appears to be resolved with the latest YOLOv5 code and the most recent stable PyTorch build (1.12 as of today), it appears to still exist in the most recent PyTorch *nightly* build (version 1.13.0. as of today), regardless of whether I use a current or old version of YOLOv5.  This is surprising, but not impossible; maybe this was fixed as a one-off prior to releasing PyTorch 1.12.  I'm crossing my fingers and hoping that when version 1.13 is released for real, this issue will (still) be fixed.
